backend/logic/discount/index.js

```diff
@@ -32,6 +32,7 @@ const validateDiscount = async (code, shop) => {
   }
 
   if (!code) {
+    // Discounts of type 'payment' won't have a code
```

Comment:
Wonder if it would be worth assigning an internal code (like 'PAYMENT') at creation tie to those automatic discounts as opposed to leaving that column empty? Might make it easier when we look at the data. Not a must-have. Just a thought.

Reply:
We should also add a `discountType != 'payment'` constraint to where clause if we end up doing this. 
Also, it will make the code `PAYMENT` a reserved one